src/styles/block/stars.scss

```diff
@@ -0,0 +1,26 @@
+.stars {
+  display: flex;
+  list-style: none;
+}
+
+.stars__star {
+  background-image: url(../../src/images/star-active.svg);
+  width: 16px;
+  height: 16px;
+  margin-right: 4px;
+  background-repeat: no-repeat;
+  background-position: center;
+}
+
+.stars__star:last-child {
+  margin-right: 0;
+}
+
+.stars--0 > .stars__star:nth-child(1n + 0),
+.stars--1 > .stars__star:nth-child(1n + 2),
+.stars--2 > .stars__star:nth-child(1n + 3),
+.stars--3 > .stars__star:nth-child(1n + 4),
+.stars--4 > .stars__star:nth-child(1n + 5),
+.stars--5 > .stars__star:nth-child(1n + 6) {
+  background-image: url(../../src/images/star.svg);
+}
```

Comment:
did you try to use [cecles](https://sass-lang.com/documentation/at-rules/control/for) here?